parity/run.rs

```diff
@@ -697,8 +692,8 @@ pub fn execute_impl(cmd: RunCmd, can_restart: bool, logger: Arc<RotatingLogger>)
 	if let Some(filter) = connection_filter {
 		service.add_notify(filter);
 	}
-	private_provider.add_notify(chain_notify.clone());
-	service.add_notify(private_provider.clone());
+	private_tx_provider.add_notify(chain_notify.clone());
+	service.add_notify(private_tx_provider.clone());
```

Comment:
this is utterly bad... Currently there is no way to prove in a test that this notifications has been set. Tbh I don't even know how to change this without significant changes to our api

Reply:
IMHO it's fine, you cannot prove many things with unit tests, that's why sometimes you write integration ones. That said,  the whole plumbing in `parity/run.rs` (especially since it's duplicated for full and light client) is currently non testable, and we should at least try to make it testable or have a framework for even more high-level (hive-like) tests (run couple of instances of parity).